### ISSUE TITLE
Make "prefer-destructuring" fix satisfy "object-curly-spacing"

### ISSUE THREAD

**Issue by galabra** [eslint/eslint#12735]
**What rule do you want to change?**
prefer-destructuring

**Does this change cause the rule to produce more or fewer warnings?**
Doesn't change the warnings amount.

**How will the change be implemented? (New option, new default behavior, etc.)?**
Just change the fix function and a few tests to match the new fixing pattern.

**Please provide some example code that this change will affect:**

```js
var foo = object.foo;
var foo = object.bar.foo;
```

**What does the rule currently do for this code?**
Changes the variable definition into the preferable destructing format.

**What will the rule do after it's changed?**
Once the rule's fix is applied, it will fix not only the "prefer-destructuring" issue, but also the "object-curly-spacing" rule, by adding spaces before and after the defined variable's name.
For example, `{ foo }` instead of `{foo}`

**Are you willing to submit a pull request to implement this change?**
Of course.

**Comment by ljharb** [comment]
Generally, one rule doesn't have any access to another rule's configuration - how would prefer-destructuring know how object-curly-spacing is set up?

The CLI uses multiple passes of autofix (editors don't, tho, i think), so this should already happen on the command line.

**Comment by galabra** [comment]
Currently auto-fixing **prefer-destructuring** results in the `{foo}` format.
You're right, eventually the **object-curly-spacing** rule will kick in and fix it to `{ foo }`, but I think it would make more sense to fix **prefer-destructuring** directly to the preferred pattern.

**Comment by ljharb** [comment]
Right, but there's no way for it to know the preferred pattern, since rules can't access other rules' configuration by design.

**Comment by galabra** [comment]
What I suggest is that the fix for prefer-destructing would be { foo }
rather than {foo}

**Comment by ljharb** [comment]
That would just cause the same problem for those who don’t want spaces inside their curly braces.

**Comment by galabra** [comment]
Well, I assumed that if there's a rule that specifically modifies curly
brackets' spacing, then it makes sense to apply it also here.
If this isn't the case, then unfortunately this issue can be closed.

‫בתאריך יום ו׳, 3 בינו׳ 2020 ב-18:44 מאת ‪Jordan Harband‬‏ <‪
notifications@github.com‬‏>:‬

**Comment by kaicataldo** [comment]
@galabra As @ljharb mentions above, rules are intended to be isolated and small in scope by design. Each rule should fix only what it is trying to enforce. In this case, `object-curly-spacing` is already a fixable rule and can handle its own autofixing.
